c10/util/typeid.h

```diff
@@ -1,30 +1,17 @@
 #pragma once
 
 #include <atomic>
-#include <cassert>
-#include <complex>
 #include <cstdlib>
 #include <memory>
 #include <mutex>
 #include <type_traits>
-#include <unordered_map>
-#include <unordered_set>
 #include <vector>
-#ifdef __GXX_RTTI
-#include <typeinfo>
```

Comment:
Okay, pretty sure this is a bug, __GX_RTTI likely indicates typeinfo is needed. You should probably add the appropiate IWYU comment to prevent it from getting removed in the future.

Reply:
<typeinfo> provides 3 classes:
```
namespace std {
  class type_info;
  class bad_cast;
  class bad_typeid;
}
```
However, none is used in typeid.h or typeid.cpp. Not sure why the header is needed...

Reply:
After greping "__GXX_RTTI" on c10, it is likely that including ```<typeinfo>``` in  typeid.h is outdated and I add the guarding in Type.h as can be seen in